csharp/src/Ice/StreamWrapper.cs

```diff
@@ -18,24 +18,22 @@ namespace IceInternal
     // For output streams, we use a different strategy:
     // Slice sequences are encoded on the wire as a count of elements, followed
     // by the sequence contents. For arbitrary .NET classes, we do not know how
-    // big the sequence that is eventually written will be. To avoid excessive
-    // data copying, this class maintains a private _bytes array of 254 bytes and,
-    // initially, writes data into that array. If more than 254 bytes end up being
-    // written, we write a dummy sequence size of 255 (which occupies five bytes
-    // on the wire) into the stream and, once this class is disposed, patch
-    // that size to match the actual size. Otherwise, if the _bytes buffer contains
+    // big the sequence will be. To avoid excessive data copying, this class maintains
+    // a private _bytes array of 254 bytes and, initially, writes data into that array.
+    // If more than 254 bytes end up being written, we write a dummy sequence size of 255
+    // (which occupies five bytes on the wire) into the stream and, once this class is disposed,
+    // patch that size to match the actual size. Otherwise, if the _bytes buffer contains
     // fewer than 255 bytes when this class is disposed, we write the sequence size
     // as a single byte, followed by the contents of the _bytes buffer.
-    //
-
-    public class OutputStreamWrapper : Stream
+    internal class OutputStreamWrapper : Stream
     {
-        public OutputStreamWrapper(Ice.OutputStream s)
+        internal OutputStreamWrapper(OutputStream s)
         {
             _s = s;
             _bytes = new byte[254];
             _pos = 0;
             _length = 0;
+            _spos = s.Tail;
         }
 
         public override int Read(byte[] buffer, int offset, int count)
```

Comment:
I think we should override the span overloads and assert / throw NotSupportedException in the byte overloads.
See https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/IO/Stream.cs#L847 and https://stackoverflow.com/questions/50078640/spant-and-streams